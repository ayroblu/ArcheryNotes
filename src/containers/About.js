import React, { Component } from 'react'
import {
  StyleSheet,
  Text,
  View
} from 'react-native'
import VersionNumber from 'react-native-version-number'

import {textStyle} from '../config'
import * as theme from '../theme'

class About extends Component {
  componentWillMount(){
    this.props.navigator.setTitle({title:'About'})
  }
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>Archery Notes App</Text>
        <Text style={styles.text}>
          Version: {VersionNumber.appVersion}, (Build: {VersionNumber.buildVersion})
        </Text>
        <Text style={styles.text}>Built by Ben Lu</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.background,
  },
  text: {
    ...textStyle
  },
})

export default About
