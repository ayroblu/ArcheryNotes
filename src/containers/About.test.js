import 'react-native'
import React from 'react'
import About from './About'
jest.mock('react-native-version-number', ()=>({
  appVersion: '1'
, buildVersion: '1.1'
}))

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders correctly', ()=>{
  const props = {
    navigator: {
      setTitle(){
        console.log('set title called')
      }
    }
  }
  const tree = renderer.create(
    <About {...props} />
  )
})
