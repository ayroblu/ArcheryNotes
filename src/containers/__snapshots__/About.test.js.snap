// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fafafa",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#555",
        "fontSize": 18,
        "fontWeight": "300",
      }
    }
  >
    Archery Notes App
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#555",
        "fontSize": 18,
        "fontWeight": "300",
      }
    }
  >
    Version: 
    1
    , (Build: 
    1.1
    )
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#555",
        "fontSize": 18,
        "fontWeight": "300",
      }
    }
  >
    Built by Ben Lu
  </Text>
</View>
`;
